src/java/org/apache/cassandra/tools/NodeTool.java

```diff
@@ -405,10 +415,12 @@ private NodeProbe connect()
                     nodeClient = nodeProbeFactory.create(host, parseInt(port));
                 else
                     nodeClient = nodeProbeFactory.create(host, parseInt(port), username, password);
+
+                nodeClient.setConsoleOutputProvider(consoleOutputProvider);
```

Comment:
OK with either. I will update the create method then.  
(mutate the ref, so the change is slightly less)

Reply:
I take it back. If also changing the `NodeProbe` constructors, it will touch other 8+ files. So I'd prefer to have a separate `setConsoleOutputProvider()` method that alters at will.